### PR TITLE
TGBOT-100 Create customer service

### PR DESCRIPTION
**Pull request by Sav4ykVladzislau**
https://schneidervs.atlassian.net/browse/TGBOT-100

### CODE REVIEW

File: stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerController.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.tutrit.stoservice.service.CustomerService;
+
+public class CustomerController implements CommandController {
+
+
+    private static final Command command = Command.SAVE_NEW_CUSTOMER;
+    public void doCommand(Request request, Response response) {
+        CustomerService.saveCustomer();
+        response.setResponse(String.format("save new customer", request.getCommand()));
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerController.java:11]: I think that String.format() not needed here

File: stoservice/src/main/java/com/tutrit/stoservice/dispatcher/RequestDispatcher.java [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
 
+import static com.tutrit.stoservice.context.ApplicationContext.get;
+
 public class RequestDispatcher {
 
     public static Logger logger = LoggerFactory.getLogger("Dispatcher");
 
     public void doDispatch(String userInput) {
-        Request request = new Request(userInput);
+        Request request = new Request(get(ParseCommand.class).getCommand(userInput));
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/dispatcher/RequestDispatcher.java:16]: I don't think it's necessary to take ParseCommand.class out of context. Methods of ParseCommand.class can be made static

File: stoservice/src/main/java/com/tutrit/stoservice/service/CustomerService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import com.tutrit.stoservice.repository.CustomerRepository;
+import com.tutrit.stoservice.utility.ParseCommand;
+
+import java.util.*;
+
+public class CustomerService {
+
+    public static Map<String, String> parseBody() {
+        List<Integer> positions = new ArrayList<>();
+        String body = ApplicationContext.get(ParseCommand.class).getBody();
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/service/CustomerService.java:14]: I don't think it's necessary to take ParseCommand.class out of context. Methods of ParseCommand.class can be made static

File: stoservice/src/main/java/com/tutrit/stoservice/service/MenuService.java [hunk 4]
```diff
@@ -1,11 +1,12 @@
 package com.tutrit.stoservice.service;
 
-import com.tutrit.stoservice.controller.RequestDispatcher;
+
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/service/MenuService.java:3]: It's just a blank line :)

File: stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerServiceController.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.tutrit.stoservice.controller;
+
+public class CustomerServiceController {
+    public void doCommand(Request request, Response response) {
+
+        response.setResponse(String.format("create a new customer", request.getCommand()));
+    }
+}
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerServiceController.java:8]: What this class is for?

**Review by yurasava — DISMISSED** [review]

File: stoservice/src/main/java/com/tutrit/stoservice/service/CustomerService.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.UUID;
+
+
+import static com.tutrit.stoservice.service.ParseBody.parseBody;
+
+public class CustomerService {
+    public static void saveCustomer(Customer customer) {
+        customer.setId(UUID.randomUUID().toString());
+        customer.setName(parseBody().get("n"));
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/service/CustomerService.java:15]: I would take the data from what comes into the method, not from a third-party class.

**Review by schneidervs — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -22,4 +22,7 @@ tg-bot/src/main/resources/application.properties
 
 bot.username={bot.username}
 
-bot.token={bot.token}
\ No newline at end of file
+bot.token={bot.token}
+
+Format a command with save Customer instance:
+"new customer - data new customer -data -n Vasil -c Brest -ph +375293845674 -e vasil@gmail.com"
```
Comment (schneidervs) [inline README.md:28]: I think there's an extra space here.

File: stoservice/src/main/java/com/tutrit/stoservice/context/ApplicationContextLoader.java [hunk 8]
```diff
@@ -24,10 +28,12 @@ public static void run(){
         put(ExitController.class, new ExitController());
         put(HelpController.class, new HelpController());
         put(NotACommandController.class, new NotACommandController());
+        put(CustomerController.class, new CustomerController());
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/context/ApplicationContextLoader.java:31]: load all dependencies for CustomerController

File: stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerController.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import com.tutrit.stoservice.bean.Customer;
+import com.tutrit.stoservice.service.CustomerService;
+
+public class CustomerController implements CommandController {
+
+
+    private static final Command command = Command.SAVE_NEW_CUSTOMER;
+    public void doCommand(Request request, Response response) {
+        CustomerService.saveCustomer(new Customer());
+        response.setResponse("save new customer");
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerController.java:12]: "saved"

File: stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerController.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.tutrit.stoservice.bean.Customer;
+import com.tutrit.stoservice.service.CustomerService;
+
+public class CustomerController implements CommandController {
+
+
+    private static final Command command = Command.SAVE_NEW_CUSTOMER;
+    public void doCommand(Request request, Response response) {
+        CustomerService.saveCustomer(new Customer());
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/controller/CustomerController.java:11]: Use service from context

**Pull request closed without merging**